### PR TITLE
XML Schema test

### PR DESCRIPTION
**Pull request by Yurasb**
1. Chages to url
2. Adding test for XML validation with XSD schema
3. Adding XSD schema for valid case

### CODE REVIEW

File: test_cases/test_cases.py [hunk 1]
```diff
@@ -1,121 +1,31 @@
 # -*- coding: utf-8 -*-
```
Comment (v1-wizard) [inline test_cases/test_cases.py]: Empty line between utf-8 comment and imports.

File: test_cases/test_cases.py [hunk 2]
```diff
@@ -1,121 +1,31 @@
...
-         '  (function(i,s,o,g,r,a,m){i[\'GoogleAnalyticsObject\']=r;i[r]=i[r]||function(){\n'
-         '  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),\n'
-         '  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)\n'
-         '  })(window,document,\'script\',\'https://www.google-analytics.com/analytics.js\',\'ga\');\n\n'
-         '  ga(\'create\', \'UA-12158002-18\', \'auto\');\n'
-         '  ga(\'send\', \'pageview\');\n}\n'
-         '</script>\n'
-         '</body>\n'
-         '</html>\n'
+    xsd=open('../test_cases/valid_case.xsd', 'r')
```
Comment (v1-wizard) [inline test_cases/test_cases.py]: Could you change xsd as file to xsd_path and hide in case object open(...)?

File: tests/test_headers.py [hunk 3]
```diff
@@ -1,55 +1,40 @@
...
-
 def test_date(headers):
     assert 'Date' in headers
 
 
-def test_expires(headers):
-    assert 'Expires' in headers
-
-
 def test_content_encoding(headers, case):
```
Comment (v1-wizard) [inline tests/test_headers.py]: Tricky. Use case as dependency to response and response as dependency to headers and use case as single fixture. https://www.python.org/dev/peps/pep-0020/#id3 - "Explicit is better than implicit."

**Review by v1-wizard — CHANGES_REQUESTED** [review]
Let's talk about? =)

File: test_cases/test_cases.py (hunk 2 above)
Comment (Yurasb) [reply test_cases/test_cases.py]: Something like that?

`class Case(object):
    def __init__self, xsd_path):
        self.xsd = open(xsd_path, 'r')
........
    xsd_path='../test_cases/valid_case.xsd'`

Comment (v1-wizard) [reply test_cases/test_cases.py]: Yes. Also you could read about with statement. https://docs.python.org/3/reference/compound_stmts.html#the-with-statement https://docs.python.org/2/tutorial/inputoutput.html

> It is good practice to use the with keyword when dealing with file objects. This has the advantage that the file is properly closed after its suite finishes, even if an exception is raised on the way. It is also much shorter than writing equivalent try-finally blocks:

File: main.py [hunk 4]
```diff
@@ -1,15 +1,11 @@
...
 with open('../config.yaml', 'r') as stream:
     configuration = yaml.load(stream)
 
-url = configuration['protocol'] +\
-      '://' +\
-      configuration['hostname'] +\
-      ':' +\
-      configuration['port'] +\
-      '/#map='
+url = '{}://{}:{}/api/{}/map?bbox='.format(
```
Comment (v1-wizard) [inline main.py:8]: You could do it like a {protocol}://{host}:{port}/api/{api_version} but this is fine too.

Comment (v1-wizard) [reply main.py:8]: .format(**kwarg) if in configuration only 4 your keys.

File: test_cases/cases.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        api_url = '{}{}'.format(url, self.url_parameters)
+        return api_url
+
+
+valid_case_with_objects = Case(
+    url_params=u'31.17268681526184,52.56401245027321,'
+               u'31.17742896080017,52.56634718820995',
+    status_code=200,
+    headers={
+        'Cache-Control': 'private, max-age=0, must-revalidate',
```
Comment (v1-wizard) [inline test_cases/cases.py:27]: Do you really need all this headers for checking?

File: tests/__init__.py [hunk 6]
```diff
@@ -0,0 +1,9 @@
+# -*- coding: utf-8 -*-
```
Comment (v1-wizard) [inline tests/__init__.py:1]: blank line.

File: tests/test_headers.py [hunk 7]
```diff
@@ -1,55 +1,47 @@
 # -*- coding: utf-8 -*-
+import requests
```
Comment (v1-wizard) [inline tests/test_headers.py:2]: blank line

File: tests/conftest.py [hunk 8]
```diff
@@ -1,34 +1,9 @@
...
-from main import url
-from test_cases.test_cases import valid_case
+from test_cases.cases import test_cases
 
 
-@pytest.fixture(scope='function')
-def case(request):
-    case = valid_case
+@pytest.fixture(scope='function', params=range(2))
+def get_case(request):
```
Comment (v1-wizard) [inline tests/conftest.py:7]: Maybe context?

File: test_cases/cases.py (hunk 5 above)
Comment (v1-wizard) [reply test_cases/cases.py:27]: Let's talk about.

**Review by v1-wizard — APPROVED** [review]

**Pull request merged**